src/CalculatorUnitTests/RationalTest.cpp

```diff
@@ -0,0 +1,226 @@
+            res = Mod(Rational(Number(1, 0, { 23242 }), Number(1, 0, { 2 })), Rational(Number(1, 0, { 0 }), Number(1, 0, { 23 })));
+            VERIFY_ARE_EQUAL(res, 11621);
+        }
+
+        TEST_METHOD(TestModuloRational)
+        {
+            // Test with rational numbers
+            auto res = Mod(Rational(Number(1, 0, { 250 }), Number(1, 0, { 100 })), Rational(89));
+            VERIFY_ARE_EQUAL(res.ToString(10, FMT_FLOAT, 8), L"2.5");
+            res = Mod(Rational(Number(1, 0, { 3330 }), Number(1, 0, { 1332 })), Rational(1));
+            VERIFY_ARE_EQUAL(res.ToString(10, FMT_FLOAT, 8), L"0.5");
+            res = Mod(Rational(Number(1, 0, { 12250 }), Number(1, 0, { 100 })), Rational(10));
+            VERIFY_ARE_EQUAL(res.ToString(10, FMT_FLOAT, 8), L"2.5");
+            res = Mod(Rational(Number(-1, 0, { 12250 }), Number(1, 0, { 100 })), Rational(10));
+            VERIFY_ARE_EQUAL(res.ToString(10, FMT_FLOAT, 8), L"7.5");
+            res = Mod(Rational(Number(-1, 0, { 12250 }), Number(1, 0, { 100 })), Rational(-10));
+            VERIFY_ARE_EQUAL(res.ToString(10, FMT_FLOAT, 8), L"-2.5");
+            res = Mod(Rational(Number(1, 0, { 12250 }), Number(1, 0, { 100 })), Rational(-10));
+            VERIFY_ARE_EQUAL(res.ToString(10, FMT_FLOAT, 8), L"-7.5");
+            res = Mod(Rational(Number(1, 0, { 1000 }), Number(1, 0, { 3 })), Rational(1));
+            VERIFY_ARE_EQUAL(res.ToString(10, FMT_FLOAT, 8), L"0.33333333");
+            res = Mod(Rational(Number(1, 0, { 1000 }), Number(1, 0, { 3 })), Rational(-10));
+            VERIFY_ARE_EQUAL(res.ToString(10, FMT_FLOAT, 8), L"-6.6666667");
+            res = Mod(Rational(834345), Rational(Number(1, 0, { 103 }), Number(1, 0, { 100 })));
+            VERIFY_ARE_EQUAL(res.ToString(10, FMT_FLOAT, 8), L"0.71");
+            res = Mod(Rational(834345), Rational(Number(-1, 0, { 103 }), Number(1, 0, { 100 })));
+            VERIFY_ARE_EQUAL(res.ToString(10, FMT_FLOAT, 8), L"-0.32");
+        }
+
+        TEST_METHOD(TestRemainerOperandsNotModified)
+        {
+            //Verify results but also check that operands are not modified
+            Rational rat25(25);
+            Rational ratminus25(-25);
+            Rational rat4(4);
+            Rational ratminus4(-4);
+            Rational res = rat25 % rat4;
+            VERIFY_ARE_EQUAL(res, 1);
+            VERIFY_ARE_EQUAL(rat25, 25);
+            VERIFY_ARE_EQUAL(rat4, 4);
+            res = rat25 % ratminus4;
+            VERIFY_ARE_EQUAL(res, 1);
+            VERIFY_ARE_EQUAL(rat25, 25);
+            VERIFY_ARE_EQUAL(ratminus4, -4);
+            res = ratminus25 % ratminus4;
+            VERIFY_ARE_EQUAL(res, -1);
+            VERIFY_ARE_EQUAL(ratminus25, -25);
+            VERIFY_ARE_EQUAL(ratminus4, -4);
+            res = ratminus25 % rat4;
+            VERIFY_ARE_EQUAL(res, -1);
+            VERIFY_ARE_EQUAL(ratminus25, -25);
+            VERIFY_ARE_EQUAL(rat4, 4);
+        }
+
+        TEST_METHOD(TestRemainerInteger)
```

Comment:
nit: Typo: `TestRemainderInteger`. Looks like it applies to other functions with `Remainder` in the name

Reply:
Tip: Never code while watching news about Brexit!